File: qa-portal/frontend/lib/core/models/bug.dart
class AppBug {
  final String id;
  final String executionResultId;
  final String title;
  final String description;
  final String severity;
  final String status;
  final String? stepsToReproduce;
  final String? createdAt;

  const AppBug({
    required this.id,
    required this.executionResultId,
    required this.title,
    required this.description,
    required this.severity,
    required this.status,
    this.stepsToReproduce,
    this.createdAt,
  });

  factory AppBug.fromJson(Map<String, dynamic> json) {
    return AppBug(
      id: json['id'] as String,
      executionResultId: json['execution_result_id'] as String,
      title: json['title'] as String,
      description: json['description'] as String,
      severity: json['severity'] as String,
      status: json['status'] as String,
      stepsToReproduce: json['steps_to_reproduce'] as String?,
      createdAt: json['created_at'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'execution_result_id': executionResultId,
      'title': title,
      'description': description,
      'severity': severity,
      'status': status,
      'steps_to_reproduce': stepsToReproduce,
      'created_at': createdAt,
    };
  }
}

File: qa-portal/backend/app/models/bug.py
from sqlalchemy import Column, String, Text, DateTime, ForeignKey, Enum as SAEnum
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime, timezone
import uuid
import enum

from app.database import Base


class BugSeverity(str, enum.Enum):
    critical = "critical"
    high = "high"
    medium = "medium"
    low = "low"


class BugStatus(str, enum.Enum):
    open = "open"
    in_progress = "in_progress"
    closed = "closed"


class Bug(Base):
    __tablename__ = "bugs"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    execution_result_id = Column(UUID(as_uuid=True), ForeignKey("test_execution_results.id", ondelete="CASCADE"), nullable=False)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    steps_to_reproduce = Column(Text, nullable=True)
    severity = Column(SAEnum(BugSeverity), nullable=False, default=BugSeverity.medium)
    status = Column(SAEnum(BugStatus), nullable=False, default=BugStatus.open)
    external_id = Column(String(100), nullable=True)  # reservado para Jira/Azure DevOps
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

    execution_result = relationship("TestExecutionResult", back_populates="bugs")
    creator = relationship("User", foreign_keys=[created_by])
